util/win11_setup.ps1

```diff
@@ -0,0 +1,374 @@
+
+    if ($wingetInstallOutput -match "No package found") {
+        Write-Host "ERROR. Package not found."
+        return $false
+    }
+    
+    if ($wingetInstallOutput -match "Successfully installed") {
+        Write-Host "OK. Installed $PackageName via winget."
+        return $true
+    }
+
+    if ($wingetInstallOutput -match "No newer package versions are available") {
+        Write-Host "OK. $PackageName already installed."
+        return $true
+    }
+
+    Write-Host $wingetInstallOutput
+    Write-Host "ERROR. Unknown install status"
+
+    return $false
+}
+
+############
+#  Script  #
+############
+
+Write-Host "Checking critical verions..."
+
+$anyInstallPending = $false
+$canInstallMissingItems = $true
+$installWsl2 = $false
+$installPython = $false
+$installGit = $false
+$installUsbipd = $false
+$installWinget = $false
+
+if (!(Confirm-wslIsInstalled) -or !(Confirm-wslCoreVersion)) {
+    $installWsl2 = $true
+    $anyInstallPending = $true
+}
+
+if (!(Confirm-wslKernelVersion)) {
+    Write-Host "CONFLICT. WSL kernel version is too old. Script cannot resolve."
+    $installWsl2 = $false
+    $canInstallMissingItems = $false
+    $anyInstallPending = $true
+}
+
+if (!(Confirm-Python3)) {
+    $installPython = $true
+    $anyInstallPending = $true
+}
+
+if (!(Confirm-gitIsInstalled)) {
+    $installGit = $true
+    $anyInstallPending = $true
+}
+
+if (!(Confirm-usbipd)) {
+    $installUsbipd = $true
+    $anyInstallPending = $true
+}
+
+if (!(Confirm-winget)) {
+    $installWinget = $true
+    $anyInstallPending = $true
+}
+
+Write-Host ""
+Write-Host ""
+
+if ($anyInstallPending) {
+    Write-Host "There are dependencies to install."
+    if ($installWsl2) {
+        Write-Host "`tWSL2"
+    }
+
+    if ($installPython) {
+        Write-Host "`tPython"
+    }
+
+    if ($installGit) {
+        Write-Host "`tGit"
+    }
+
+    if ($installUsbipd) {
+        Write-Host "`tUsbipd"
+    }
+
+    Write-Host ""
+} else {
+    Write-Host "All dependencies valid."
+    Write-Host ""
+
+    Exit 0
+}
+
+
+
+if (!($install)) {
+    Write-Host "Use the -install flag to install missing items."
+    Write-Host ""
+
+    Exit 0
+}
+
+##################
+#  Install Step  #
+##################
+
+$installStepFailed = $false
+
+if ($installWsl2) {
+    Write-Host "UNWILLING. There are too many pitfalls to automate this easily."
+    Write-Host "Please install a WSL2 Ubuntu LTS distro from the store and re-run the script."
+    # NOTE: Will Stuckey has prior work here, but the hypervisor mode failures are esoteric
+    # and quite a pain to detect and support for arbitrary users. Make them do this manually.
+    Exit 1
+}
+
+if ($installWinget) {
+    Write-Host "UNABLE. Windows 11 should come with winget."
+    Write-Host "Unable to proceed with package installations."
+    Exit 1
+}
+
+if ($installPython) {
+    $success = Install-wingetPackage Python.Python.3.11
```

Comment:
Is there a reason this version number is different from the min version up top?

Reply:
I think this will run on old stuff, but i figured don't install 3.9. Its like a year away from EoL? I could bump the min version to 10 or 11